src/ekat/util/ekat_string_utils.cpp

```diff
@@ -58,6 +59,73 @@ std::string upper_case (const std::string& s) {
   return s_up;
 }
 
+double jaro_similarity (const std::string& s1, const std::string& s2) {
+  // Two equal strings always have similarity of 1, regardless of whether they are empty or not
+  if (s1==s2) {
+    return 1;
+  }
+
+  const int len1 = s1.size();
+  const int len2 = s2.size();
+  const int max_dist = std::max(len1,len2)/2 - 1;
+
+  double matches = 0;
+  std::vector<int> s1_matches(len1,0);
+  std::vector<int> s2_matches(len2,0);
+  for (int i=0; i<len1; ++i) {
+    for (int j=std::max(0,i-max_dist); j<std::min(len2,i+max_dist+1); ++j) {
+      if (s1[i]==s2[j] && s2_matches[j]==0) {
+        matches += 1.0;
+        s1_matches[i] = 1;
+        s2_matches[j] = 1;
+        break;
+      }
+    }
+  }
+
+  // If no matches, we're done
+  if (matches==0) {
+    return 0;
+  }
+
+  // There are matches. Count transpositions
+  double transp = 0;
+  for (int i=0, k=0; i<len1; ++i) {
+    if (s1_matches[i]==1) {
+      while (s2_matches[k]==0) {
+        ++k;
+      }
+      if (s1[i]!=s2[k]) {
+        transp += 0.5;
+      }
+      ++k;
+    }
+  }
+
+  return ( matches/len1 + matches/len2 + (matches-transp)/matches ) / 3.0;
+}
+
+double jaro_winkler_similarity (const std::string& s1, const std::string& s2,
```

Comment:
Same request here--it'd just be nice to get a brief description of this algorithm, typical output values for some simple input, and maybe contrast this algorithm with the Jaro algorithm above.